src/environment.js

```diff
@@ -93,6 +93,8 @@ export type Binding = {
   environment: EnvironmentRecord,
   name: string,
   isGlobal: boolean,
+  // bindings that are assigned to inside loops with abstract termination conditions need temporal locations
+  phiNode?: AbstractValue,
```

Comment:
Is phiNode's lifetime spanning across widenBindings() invocation session? If not, since phiNode is only used inside widenBindings(), maybe we could create a Map<Binding, AbstractValue> map inside widenBindings to temporal store these values? By doing this, we do not need to expose "phiNode" outside of widenBindings() function and these values can be released after widenBindings() call.

Reply:
The point is exactly that the same phi node should be used across all fix point iterations.